### PR TITLE
[FEATURE] Add AssetHandling Chapter

### LINKED ISSUE #959: Feature: 90522 - Introduce AssetCollector (10.3)
| Name        | Value     |
| -------------- | ----------- |
| Type:         |  Feature |
| Merged in: | 10.3       |
| Gerrit:        | [63327](https://review.typo3.org/c/Packages/TYPO3.CMS/+/63327) |
| Changelog: | [Feature: 90522 - Introduce AssetCollector](https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.3/Feature-90522-IntroduceAssetCollector.html) |

Resources:

* usetypo3: https://usetypo3.com/asset-collector.html

Related: #845 (general changelog issue for 10.3)

### PR DESCRIPTION
**Pull request by susannemoog**
See Core Feature 90522
See #845
Resolves #959

### CODE REVIEW

File: Documentation/ApiOverview/Assets/Index.rst [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+Examples
+--------
+
+Add a JavaScript file to the collector with script attribute :code:`data-foo="bar"`:
+
+.. code-block:: php
+
+    GeneralUtility::makeInstance(AssetCollector::class)
+       ->addJavaScript('my_ext_foo', 'EXT:my_ext/Resources/Public/JavaScript/foo.js', ['data-foo' => 'bar']);
```
Comment (mbrodala) [inline Documentation/ApiOverview/Assets/Index.rst:110]: We should also explain what happens if the same asset is registered multiple times with different attributes.

Comment (jonaseberle) [reply Documentation/ApiOverview/Assets/Index.rst:110]: Good point. 
It merges with the existing assets - https://github.com/TYPO3/TYPO3.CMS/blob/master/typo3/sysext/core/Classes/Page/AssetCollector.php#L77.

For more control the AssetRenderer events https://docs.typo3.org/c/typo3/cms-core/master/en-us/Changelog/10.4/Feature-90899-IntroduceAssetPreRenderingEvents.html#changelog-feature-90899-introduceassetprerenderingevents shall be used.

Comment (susannemoog) [reply Documentation/ApiOverview/Assets/Index.rst:115]: Thank you both for your comments (and the links), I added an explanation of the merging as well as the description of the events and cross-links to them.

File: Documentation/ApiOverview/Hooks/Events/Core/Page/AssetRendererEvent.rst.txt [hunk 2]
```diff
@@ -0,0 +1,112 @@
+The :php:`AssetRenderer` is amended by two events which allow post-processing of
+:php:`AssetCollector` assets.
+
+* :php:`\TYPO3\CMS\Core\Page\Event\BeforeJavaScriptsRenderingEvent`
+* :php:`\TYPO3\CMS\Core\Page\Event\BeforeStylesheetsRenderingEvent`
+
+The events are fired exactly once for every combination of
```
Comment (flicstar) [inline Documentation/ApiOverview/Hooks/Events/Core/Page/AssetRendererEvent.rst.txt:7]:
```suggestion
The events are fired once for every combination of
```
Do we need "exactly" here?

**Comment by flicstar** [comment]
I read through this and it seems fine to me.

**Review by mbrodala — APPROVED** [review]

**Pull request merged**